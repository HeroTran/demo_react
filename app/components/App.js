import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import '../app.scss';

const App = ({ children }) =>
    <div>
        <h1 className="app">Magazine</h1>
        { children }
        <footer>
            <Link to="/">Home</Link>
            <Link to="/issue">Issue</Link>
            <Link to="/site/121">Site</Link>
        </footer>
    </div>;

App.propTypes = {
    children: PropTypes.object
};

export default App;
