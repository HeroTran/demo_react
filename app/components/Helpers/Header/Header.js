import React from 'react'
import './Header.scss'

export default class Header extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div  className="Header">
        <div className="Header-search">
            <form>
                <input type="search" className="Header-searchInput" placeholder = "Search for Magazine"/>
                <i className="Header-searchIcon material-icon md-24 dask">search</i>
            </form>
        </div>
      </div>
    );
  }
}
