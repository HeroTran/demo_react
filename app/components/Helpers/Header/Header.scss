.Header{
	background-color: #fff;
	height: 60px;
	width: 100%;	
	box-sizing: border-box;
	display: block;
	flex-wrap: nowrap;
	background-repeat: no-repeat;
	font-family: Arial, Helvetica, sans-serif;
	justify-content: space-between;
	align-self: center;
	background-image: url("/app/assets/images/logo-header.png");
	background-size: auto 30px;
	background-position: center center !important;
	.Header-search{
		float: right;
		display: block;
		margin-right: auto;
		.Header-searchInput{
			position: absolute;
			padding: 0;
			width: 24px;
			height: 60px;
			z-index: 2;
			left: auto;
			right: 40px;
			margin-right:-40px;
			font-size: 14px;
			box-shadow:none;
			background-color: transparent;
			cursor:pointer;
			color: transparent;
			&::-webkit-input-placeholder {
				color: transparent;
			}

		}
	}
	i{
		line-height: 60px;
		cursor: pointer;
		vertical-align: top;
		margin:0 0.25em;
		height: 60px;
		&:hover {
			color: #00a2e0;
			transition: all 0.4s ease-in-out;
		}
	}
}

.Header-expandableSearch {
	position: relative;
	display: inline-block;
	height: 60px;
	width: 24px;
	cursor: pointer;
	margin-right: 2.25em;
	vertical-align: top;
	color: #00a2e0;

}

input[type=search] {
	transition: all 0.4s;
	border: none;
	outline: none;
	border-radius: 0;
	padding: 0;
	-webkit-appearance: none;
	-webkit-border-radius: 0px;
	&::-webkit-search-cancel-button{
		display:none;
	}
}


.Header-searchInput:focus {
	width: 200px !important;
    right: 60px;
    color: #00a2e0 !important;
    cursor: auto;
    border-bottom: 1px solid;
    border-color: #00a2e0;
    &::-webkit-input-placeholder {
    	color: #00a2e0 !important;
    }

}


.Header-searchInput:hover + i.Header-searchIcon {
	color: #00a2e0;
	transition: color 0.4s ease-in-out;
}

.Header-searchInput.is-filled + i.Header-searchIcon {
	color: #00a2e0;
	transition: all 0.4s ease-in-out;
}

i.Header-searchIcon {
	display: inline-block;
	height: 60px;
	line-height: 60px;
	text-align: center;
	text-decoration: none;
	user-select: none;
	text-align: center;
	margin: 0;
	padding: 0;
} 