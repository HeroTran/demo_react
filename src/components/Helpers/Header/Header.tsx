import * as React from 'react'
import './Header.scss'

export default class Header extends React.Component<any,any> {
  
  render() {
    return (
      <div  className="Header">
        <div className="Header-search">
            <div className="Header-expandableSearch">
                <form>
                    <input type="search" className="Header-searchInput" placeholder = "Search for Magazine"/>
                    <i className="Header-searchIcon material-icons md-12 dark">search</i>
                </form>
            </div>
        </div>
      </div>
    );
  }
}
