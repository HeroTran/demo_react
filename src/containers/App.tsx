import * as React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import Header from '../components/Helpers/Header/Header';
import reducers from '../reducers';


interface Props extends React.Props<App> {
}

export default class App extends React.Component<Props, {}> {
  public render() {
    return (
      <div className="content-app">
        <Header />
        {this.props.children}
      </div>
    );
  }
}
