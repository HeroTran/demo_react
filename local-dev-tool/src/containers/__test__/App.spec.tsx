import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {mount, shallow} from 'enzyme';
import App from '../App';
import Header from '../../components/Helpers/Header/Header';

describe("App component", () => {
    it("renders only 1 App component", () => {
        const component = shallow(<App />);
        expect(component).toHaveLength(1);
    });
    it('renders a snapshot', () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should contains a header', () => {
        const component = mount(<App />);
        expect(component.contains(<Header />)).toEqual(true);
    })
});