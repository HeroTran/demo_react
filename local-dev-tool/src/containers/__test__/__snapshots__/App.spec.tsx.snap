// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders a snapshot 1`] = `
<div
  className="content-app"
>
  <div
    className="Header"
  >
    <div
      className="Header-Home"
    >
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="App dev Tool"
          src="/images/logo-header.png"
        />
      </a>
    </div>
  </div>
</div>
`;
